### PR TITLE
Remove onAbortRequested python callback functionality

### PR DESCRIPTION
**Pull request by jimfcarroll**
## Description

Reimplement the removal of onAbortRequested. The initial attempt was ... aborted ... here: https://github.com/xbmc/xbmc/pull/17552

## Motivation and Context

Remove deprecated functionality.

## Types of change

- [ ] **Bug fix** (non-breaking change which fixes an issue)
- [x] **Clean up** (non-breaking change which removes non-working, unmaintained functionality)
- [ ] **Improvement** (non-breaking change which improves existing functionality)
- [ ] **New feature** (non-breaking change which adds functionality)
- [ ] **Breaking change** (fix or feature that will cause existing functionality to change)
- [ ] **Cosmetic change** (non-breaking change that doesn't touch code)
- [ ] **None of the above** (please explain below)

## Checklist:

- [ ] My code follows the **[Code Guidelines](https://github.com/xbmc/xbmc/blob/master/docs/CODE_GUIDELINES.md)** of this project 
- [X] My change requires a change to the documentation, either Doxygen or wiki
- [X] I have updated the documentation accordingly
- [ ] I have read the **[Contributing](https://github.com/xbmc/xbmc/blob/master/docs/CONTRIBUTING.md)** document
- [ ] I have added tests to cover my change
- [ ] All new and existing tests passed

### CODE REVIEW

**Comment by fuzzard** [comment]
@MilhouseVH could you give this a try with your libreelec service script when you have time.

**Comment by MilhouseVH** [comment]
Will do. An initial test with a local build looks good here: http://ix.io/2iJJ

I'll include this in #`0420`.

**Comment by MilhouseVH** [comment]
Actually, it looks like kodi.bin is crashing at the very end: http://ix.io/2iJN

In the system log it has these entries:
```
Apr 19 23:04:05 NUC systemd[1]: Stopping Kodi Media Center...
Apr 19 23:04:08 NUC kernel: FDEventMonitor[1251]: segfault at ffffffffffffffe8 ip 0000000000d4da45 sp 00007f5a4ccc1a30 error 5 in kodi.bin[400000+ed2000]
Apr 19 23:04:08 NUC kernel: Code: 44 24 10 00 00 00 00 48 c7 44 24 18 00 00 00 00 e8 0c 5a 07 00 89 c3 45 39 e5 7e 04 3c 01 75 45 48 8b 75 08 4d 8b 06 44 89 e1 <48> 8b 56 e8 4d 8b 48 e8 50 ff 74 24 20 ff 74 24 20 ff 74 24 20 48
Apr 19 23:04:09 NUC kodi.sh[1238]: Segmentation fault (core dumped)
Apr 19 23:04:10 NUC kodi.sh[1238]: Crash report available at /storage/.kodi/temp/kodi_crashlog_20200419230409.log
```

Do you want the output from a debug-enabled crash log? I can get you that in a couple of hours.

**Comment by ronie** [comment]
great, thanx for the help @jimfcarroll 

mind adding the documentation bit:
https://github.com/xbmc/xbmc/commit/2bb85067d8ebde85295cc495ce941087b241ccfd#diff-1895c13c2768bd33563de46a082f235a

**Comment by jimfcarroll** [comment]
@MilhouseVH can you give me the script you think is triggering it?

**Comment by jimfcarroll** [comment]
@ronie done.

**Comment by MilhouseVH** [comment]
@jimfcarroll all I'm running is `systemctl stop kodi` to shutdown kodi. 

Unfortunately the shutdown behaviour is a little unpredictable, it would seem.

After further investigation, I don't think the crash is due to this PR, and is in fact a long standing shutdown crash issue - somewhere in logging.

So all in all, **this PR is probably OK**, as it appears to trigger add-on shutdown correctly, only for kodi to then crash elsewhere.

As I'd already tested various configurations before coming to the above conclusion, I'll post the information I collected below with logs - maybe the crash logs will be useful to someone.

If you can shed any light on the LibreELEC `dbus` issue that would be most excellent and very much appreciated. It seems to occur when the [`dbusSystemBus` object](https://github.com/LibreELEC/service.libreelec.settings/blob/7b5ef62bd17a2357fbdd43aa543194ab4f4beee5/src/oe.py#L67) goes out of scope - presumably to be garbage collected - the Python interpreter thread then hangs. Maybe it's a bug in `dbus`, we're not really sure.

----

The system I normally test with (and which crashed) has a few add-ons installed (`Netflix`, `LibreELEC Settings` - aka `service.libreelec.settings` - etc.) and that produces this debug-enabled crash log on shutdown: http://ix.io/2iKZ - `kodi.bin` seems to be crashing somewhere in logging/`spdlog`/`libfmt`.

This is most probably "normal" - `kodi.bin` crashing somewhere in logging has been an issue for a very long time, and appears to still be an issue after the `spdlog` change.

I then started testing again with a "clean" `.kodi` and now Kodi doesn't shutdown at all - it hangs on shutdown, and there is no crash. In this case, after requesting `systemd` to shutdown kodi, it had to kill `kodi.bin` after a 600 second timeout. It appears to be hanging while trying to shut down `service.libreelec.settings` (which since Python3 has a known issue with `dbus` when shutting down - some issue with the Python gc is our best guess so far): http://ix.io/2iLi - again, this isn't unusual, it's an issue since the switch to Python3, and unique to the LibreELEC Settings add-on (sadly).

Hmmm.

Next, I tested a "clean" `.kodi`, this time without the `service.libreelec.settings` addon and kodi crashed on shutdown: http://ix.io/2iLj - again, crashing somewhere in logging.

If I don't include this PR (and without `service.libreelec.settings` addon, and a clean `.kodi` etc.) then the system continues to crash in logger... sigh. http://ix.io/2iLk

So in summary, if kodi doesn't hang due to `service.libreelec.settings`, then it will most likely crash due to a segfault in logging. But none of that is due to this PR. :)

**Comment by MilhouseVH** [comment]
Not sure if this is a PR issue, or add-on issue: http://ix.io/2iZ6 - the netflix addon isn't aborting correctly and is having to be killed. Ping @CastagnaIT 
```
2020-04-21 20:51:32.602 T:1559    DEBUG <general>: CServiceAddonManager: stopping plugin.video.netflix.
2020-04-21 20:51:32.602 T:1559    DEBUG <general>: CPythonInvoker(0, /storage/.kodi/addons/plugin.video.netflix/service.py): trigger Monitor abort request
2020-04-21 20:51:37.610 T:1559    ERROR <general>: CPythonInvoker(0, /storage/.kodi/addons/plugin.video.netflix/service.py): script didn't stop in 5 seconds - let's kill it
2020-04-21 20:51:37.611 T:1559    DEBUG <general>: CServiceAddonManager: stopping plugin.video.youtube.
2020-04-21 20:51:37.611 T:1559    DEBUG <general>: CPythonInvoker(1, /storage/.kodi/addons/plugin.video.youtube/resources/lib/startup.py): trigger Monitor abort request
2020-04-21 20:51:37.611 T:1590    DEBUG <general>: [plugin.video.youtube] PlaybackMonitor active threads: ||
2020-04-21 20:51:37.614 T:1590     INFO <general>: CPythonInvoker(1, /storage/.kodi/addons/plugin.video.youtube/resources/lib/startup.py): script successfully run
2020-04-21 20:51:37.614 T:1590    DEBUG <general>: onExecutionDone(1, /storage/.kodi/addons/plugin.video.youtube/resources/lib/startup.py)
2020-04-21 20:51:37.615 T:1590    DEBUG <general>: .
2020-04-21 20:51:37.941 T:1590     INFO <general>: Python interpreter interrupted by user
2020-04-21 20:51:37.941 T:1559    DEBUG <general>: CPythonInvoker(1, /storage/.kodi/addons/plugin.video.youtube/resources/lib/startup.py): script termination took 330ms
2020-04-21 20:51:37.941 T:1559    DEBUG <general>: CServiceAddonManager: stopping script.common.plugin.cache.
2020-04-21 20:51:37.941 T:1559    DEBUG <general>: CPythonInvoker(2, /storage/.kodi/addons/script.common.plugin.cache/resources/lib/entry_point.py): trigger Monitor abort request
2020-04-21 20:51:37.942 T:1589     INFO <general>: CPythonInvoker(0, /storage/.kodi/addons/plugin.video.netflix/service.py): script aborted
2020-04-21 20:51:37.942 T:1589    ERROR <general>: CPythonInvoker(0, /storage/.kodi/addons/plugin.video.netflix/service.py): failed to set abortRequested
2020-04-21 20:51:37.942 T:1589    DEBUG <general>: onExecutionDone(0, /storage/.kodi/addons/plugin.video.netflix/service.py)
2020-04-21 20:51:37.942 T:1589    DEBUG <general>: .
2020-04-21 20:51:38.187 T:1589     INFO <general>: Python interpreter interrupted by user
```

This is a log without this PR, where Netflix aborts correctly (taking 1022ms): http://ix.io/2iZ8

**Comment by CastagnaIT** [comment]
netflix check monitor.abortRequested method to stop services loops
there is a downloadable win64 builded artifact of this PR?

**Comment by enen92** [comment]
@CastagnaIT started one manually in https://jenkins.kodi.tv/view/Automation/job/BuildMulti-PR-Manually/338/

Check the mirrors (test builds) when the job finishes: http://mirrors.kodi.tv/test-builds/windows/win64/

**Comment by jimfcarroll** [comment]
@MilhouseVH do you have a simple plugin/addon that uses the abortRequested or waitOnAbort? If not, I'll write one to test this when I get a chance.

**Comment by MilhouseVH** [comment]
@jimfcarroll unfortunately I do not - whatever test scripts I've used in the past were wiped long ago as I didn't anticipate revisiting this!

**Comment by enen92** [comment]
You can use script.tubecast if needed, is already in matrix repo. It launches an http server (bottle) and a threaded udp server in the background, and should exits gracefully

**Comment by CastagnaIT** [comment]
the build is done but not see output on mirrors

**Comment by enen92** [comment]
@CastagnaIT that's because I'm retarted, forgot to select "upload build result". Here's another one: https://jenkins.kodi.tv/view/Automation/job/BuildMulti-PR-Manually/339/

**Comment by CastagnaIT** [comment]
i have tried 5 times (with and without "Synchronize the watched status of the videos with Netflix" that add another thread loop)

but it was always finished correctly
e.g. https://paste.kodi.tv/erohewuzed.kodi

I couldn't test after a video playback because ImputStreamAddon is currently broken for API changes, and I can't get it to work, manually changing api version works for installation but it will be disabled when try to play a video...
but it shouldn't make difference

can you try running some more tests?

**Comment by MilhouseVH** [comment]
@CastagnaIT yes, it's not easy to reproduce. Maybe it was a one-off. I guess unless it becomes a frequent issue it may not be anything to worry about. Certainly I think we can get this PR merged so that it can be tested with a wider group.
